geonode/assets/utils.py

```diff
@@ -84,10 +85,38 @@ def create_link(resource, asset, link_type=None, extension=None, name=None, mime
     return link
 
 
+def create_asset(
+    owner,
+    files,
+    handler=None,
+    title=None,
+    description=None,
+    asset_type=None,
+    clone_files: bool = True,
+) -> Asset:
+    asset_handler = handler or asset_handler_registry.get_default_handler()
+    asset = None
+    try:
+        asset = asset_handler.create(
+            title=title or "Unknown",
+            description=description or asset_type or "Unknown",
+            type=asset_type or "Unknown",
+            owner=owner,
+            files=files,
+            clone_files=clone_files,
+        )
+        return asset
+    except Exception as e:
+        logger.error(f"Error creating Asset: {e}", exc_info=e)
+        if asset:
```

Comment:
if an exception is raised within the `try` block, `asset` will neevr be assigned